compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -543,13 +543,26 @@ object QuotesAndSplices {
   def getQuotedPatternTypeVariable(name: TypeName)(using Context): Option[Symbol] =
     ctx.property(TypeVariableKey).get.get(name)
 
-    /** Get the symbol for the quoted pattern type variable if it exists */
+  /** Get the symbol for the quoted pattern type variable if it exists */
   def addQuotedPatternTypeVariable(sym: Symbol)(using Context): Unit =
     ctx.property(TypeVariableKey).get.update(sym.name.asTypeName, sym)
 
-  /** Context used to type the contents of a quoted */
-  def quotePatternContext()(using Context): Context =
+  /** Context used to type the contents of a quote pattern */
+  def quotePatternContext(isTypePattern: Boolean)(using Context): Context =
     quoteContext.fresh.setNewScope
-      .addMode(Mode.QuotedPattern).retractMode(Mode.Pattern)
+      .addMode(if isTypePattern then Mode.QuotedTypePattern else Mode.QuotedExprPattern)
+      .retractMode(Mode.Pattern) // TODO do we need Mode.QuotedPattern?
       .setProperty(TypeVariableKey, collection.mutable.Map.empty)
+
+  /** Context used to type the contents of a quote pattern splice */
+  def quotePatternSpliceContext(using Context): Context =
+    spliceContext
+      .retractMode(Mode.QuotedPatternBits)
+      .addMode(Mode.Pattern)
+      .withOwner(quoteOwner(ctx))
+
+  /** Owner of the quoted pattern */
```

Comment:
Is the documentation accurate since it also does something when we're not in a quoted pattern?

Reply:
I updated the documentation and name of the definition